### PR TITLE
Feature/document publications search

### PR DESCRIPTION
**Pull request by ivanmrsulja**
## Description

Added generic advanced search query parser with AND, OR and NOT operators. Added advanced search functionality for DocumentPublication, DocumentFile and Person entities. Addes simple all field search to DocumentFile entity.

## How should this be tested?

Unit tests for new functionalities are implemented, you can try search endpoints through Postman or cURL if you want.

## Checklist

-   [x] I have tested these changes locally
-   [x] I have added tests that cover these changes (if applicable)
-   [x] I have updated the documentation (if applicable)
-   [x] I have added or updated the necessary comments in the code

### CODE REVIEW

File: src/main/java/rs/teslaris/core/service/impl/document/DocumentFileServiceImpl.java [hunk 1]
```diff
@@ -156,6 +169,46 @@ public void parseAndIndexPdfDocument(DocumentFile documentFile, MultipartFile mu
...
+            });
+            return b;
+        })))._toQuery();
+    }
+
+    private Query buildAdvancedSearchQuery(List<String> expression) {
+        var postfixExpression =
+            expressionTransformer.transformToPostFixNotation(expression);
+        return expressionTransformer.buildQueryFromPostFixExpression(postfixExpression);
+    }
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/service/impl/document/DocumentFileServiceImpl.java:210]: This code is duplicated. Can we make this method only inside one class? Not sure about expressionTransformer, do we need separate objects for two classes for some reason?

File: src/main/java/rs/teslaris/core/service/impl/person/PersonServiceImpl.java [hunk 2]
```diff
@@ -375,6 +379,16 @@ public Page<PersonIndex> findPeopleForOrganisationUnit(
             List.of(employmentInstitutionId));
     }
 
+    @Override
+    public Page<PersonIndex> searchPersonsAdvanced(SearchRequestDTO searchRequest,
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/service/impl/person/PersonServiceImpl.java:383]: probably this should be renamed to advancedSearch

**Review by chenejac — CHANGES_REQUESTED** [review]
@ivanmrsulja I have posted a couple of comments. Please check it before we merge this PR. Well done!!!

File: src/main/java/rs/teslaris/core/service/impl/document/DocumentFileServiceImpl.java (hunk 1 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/service/impl/document/DocumentFileServiceImpl.java:210]: I managed to reduce the code to one line by encapsulating expression parsing. That way if we ever change parsing technique the responsibility will be in one single method.

**Review by chenejac — CHANGES_REQUESTED** [review]
@ivanmrsulja It looks this is also ready for merging. Please, resolve merge conflicts for the sake of merging PR.

**Review by chenejac — APPROVED** [review]

**Pull request merged**